modules/modules.go

```diff
@@ -20,6 +20,9 @@ type module struct {
 
 	// is this module user visible (i.e. intended to be passed to `InitModuleServices`)
 	userVisible bool
+
+	// is the module allowed to be selected as a target
+	targetable bool
```

Comment:
Now we have two flags for each module. These flags are only used by caller, not in this package. I wonder if we should replace them with generic string "flags", and add method to check if given module has a specific flag. Then callers can assign whatever meaning to the flags that they want.

(In this design, module would be registered with list of strings, and module would just keep the list, but not interpret it in any way.)

Reply:
I think what you're describing might make sense, but seems like a bit out of scope for this simple PR, as it would probably be a breaking change which requires changes in all the users of this module management system.

Reply:
I'm not sure it's out of scope -- either we introduce new field, or generic list of flags. It's just a different way of solving the same problem. We could do it without breaking existing clients by supporting (and deprecating) existing `IsUserInvisible` method and changing `UserInvisibleModule` to be a string.